main/src/library/Reducible.flix

```diff
@@ -70,8 +70,11 @@ pub class Reducible[t: Type -> Type] {
     /// Right-associative fold of a structure.
     /// Applies `f` to a start value `s` and all elements in `t` going from right to left.
     ///
-    pub def foldRight(f: (a, b) -> b & ef, s: b, t: t[a]): b & ef =
-        Reducible.reduceRightTo(f, a -> f(a, s), t)
+    /// A `foldRight` allows early termination by not calling the continuation.
+    ///
+    pub def foldRight(f: (a, Unit -> b & ef) -> b & ef, z: b, t: t[a]): b & ef =
+        let f1 = (a, b) -> f(a, _ -> b as & ef);
```

Comment:
Are these casts necessary? Maybe so?

Reply:
I think so, but please confirm :)

Reply:
Yes, otherwise it says that `foldRight` should have effect `Pure` instead of `ef`